pkg/cmd/env/connect.go

```diff
@@ -0,0 +1,162 @@
+package env
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/coreeng/corectl/pkg/cmdutil/config"
+	"github.com/coreeng/corectl/pkg/cmdutil/userio"
+	corectlenv "github.com/coreeng/corectl/pkg/env"
+	"github.com/coreeng/corectl/pkg/gcp"
+	"github.com/coreeng/developer-platform/pkg/environment"
+	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
+)
+
+type EnvConnectOpt struct {
+	Port               int
+	Environment        string
+	RepositoryLocation string
+	ProjectID          string
+	Region             string
+	Streams            userio.IOStreams
+	Exec               corectlenv.Commander
+	gcpClient          gcp.Client
+}
+
+func connectCmd(cfg *config.Config) *cobra.Command {
+	opts := EnvConnectOpt{
+		Exec: corectlenv.NewCommand(),
+	}
+	connectCmd := &cobra.Command{
+		Use:   "connect [environment]",
+		Short: "Connect to an environment",
+		Long:  `This command allows you to connect to a specified environment.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.Streams = userio.NewIOStreams(
+				cmd.InOrStdin(),
+				cmd.OutOrStdout(),
+			)
+			return connect(opts)
+		},
+	}
+
+	connectCmd.Flags().StringVarP(
+		&opts.RepositoryLocation,
+		"repository",
+		"r",
+		cfg.Repositories.CPlatform.Value,
+		"Repository to source environments from",
+	)
+
+	connectCmd.Flags().StringVarP(
+		&opts.Environment,
+		"environment",
+		"e",
+		"",
+		"Environment to connect to",
+	)
+
+	connectCmd.Flags().IntVarP(
+		&opts.Port,
+		"port",
+		"p",
+		54808,
+		"Local port to use for connection to the cluster",
+	)
+
+	config.RegisterStringParameterAsFlag(
+		&cfg.Repositories.CPlatform,
+		connectCmd.Flags(),
+	)
+
+	return connectCmd
+}
+
+func connect(opts EnvConnectOpt) error {
+	envs, err := environment.List(environment.DirFromCPlatformRepoPath(opts.RepositoryLocation))
+	if err != nil {
+		return err
+	}
+
+	if opts.Environment == "" {
+		inputEnv := createEnvInputSwitch(envs)
```

Comment:
You should be able to just `DefaultValue: userio.AsZeroable(opts.Environment)` inside the `createEnvInputSwitch` instead of `if` on this level